corva/middleware/loader.py

```diff
@@ -0,0 +1,14 @@
+from typing import Callable, Optional
+
+from corva.models.base import BaseContext
+
+
+def loader_factory(loader: Callable, loader_kwargs: Optional[dict] = None) -> Callable:
```

Comment:
Can we simplify our interface to creating our own middleware functionality? I wouldn't want to have to implement the factory pattern every time. Can we make a simpler interface maybe?

Reply:
yes, so the current interface looks like this
```python3
def middleware(context, call_next): pass
```

the factories are used only to parametrize the middleware, any other approach can be used e.g
```python3
# approach 1

class Middleware:
    def __init__(self, param1):
        self.param1 = param1

    def __call__(self, context, call_next):
        pass


mw = Middleware(param1='')  # parametrize like this


# approach 2

def middleware(context, call_next, param1):
    pass


mw = functools.partial(middleware, param1='')  # parametrize like this


# approach 3

def middleware_factory(param1):
    def middleware(context, call_next):
        pass

    return middleware


mw = middleware_factory(param1='')  # parametrize like this
```